pkg/cloudprovider/aws.go

```diff
@@ -71,20 +72,34 @@ func (a *AWS) AssignPrivateIP(ip net.IP, node *corev1.Node) error {
 	networkInterface := networkInterfaces[0]
 	addIP := ip.String()
 	keepIPs := []*string{}
+	newIPs := []*string{}
 	if utilnet.IsIPv6(ip) {
 		for _, assignedIPv6 := range networkInterface.Ipv6Addresses {
 			if assignedIP := net.ParseIP(*assignedIPv6.Ipv6Address); assignedIP != nil && assignedIP.Equal(ip) {
 				return AlreadyExistingIPError
 			}
 			keepIPs = append(keepIPs, assignedIPv6.Ipv6Address)
 		}
-		keepIPs = append(keepIPs, &addIP)
```

Comment:
What's the point of having `keepIPs` now? we don't use it to assign the new IP anyway

Reply:
I renamed it to exsitingIPs and I added a comment above for why we need it. In go, we cant print the content of a slice of pointers, we'll get the only the pointers if we `fmt.Sprintf("%v", sliceOfPointers)`. So unless you know of something more elegant, we still must collect those IPs in a new []string{} slice